src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.LanguageServices.Implementation.LanguageServiceBrokerShim;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Interaction logic for DocumentOutlineControl.xaml
+    /// </summary>
+    internal partial class DocumentOutlineControl : UserControl, IVsCodeWindowEvents
+    {
+        private ILanguageServiceBrokerShim LanguageServiceBroker { get; }
+
+        private IThreadingContext ThreadingContext { get; }
+
+        private IVsEditorAdaptersFactoryService EditorAdaptersFactoryService { get; }
+
+        private IVsCodeWindow CodeWindow { get; }
+
+        /// <summary>
+        /// The type of sorting to be applied to the UI model in <see cref="UpdateUIAsync"/>.
+        /// </summary>
+        private SortOption SortOption { get; set; }
+
+        /// <summary>
+        /// Queue to batch up work to do to compute the UI model. Used so we can batch up a lot of events 
+        /// and only fetch the model once for every batch.
+        /// </summary>
```

Comment:
would be good to explain what hte `bool` input parameter is for.  it's not self evident to me.  

Also, it says it's computing the UI model... but is taht true?  it looks like it's computing DocumentSymoblModel, which i thought was the data-model? (but i could be wrong there).